src/Parsers/ParserDataType.cpp

```diff
@@ -27,7 +27,7 @@ class ParserDataTypeArgument : public IParserBase
     {
         ParserNestedTable nested_parser;
         ParserDataType data_type_parser;
-        ParserLiteral literal_parser;
+        ParserAllCollectionsOfLiterals literal_parser;
```

Comment:
And this is the second fix to allow complex parameters.

Reply:
One interesting thing about `ParserAllCollectionsOfLiterals` is that we would also parse expressions like `{1:'a'}` approximately as `Map(UInt8, String)`, which is not what we generally do.

Also, this parser is a bit simpler than others in the sense that we do not fill the `begin` and `end` fields of ASTLiteral, which is used at least in one place in the code. (src/Processors/Formats/Impl/ConstantExpressionTemplate.cpp:199)

I will make the fix for it soon.